controllers/managedocs_controller.go

```diff
@@ -44,72 +45,125 @@ import (
 // +kubebuilder:rbac:groups="",namespace=system,resources=secrets,verbs=get;list;watch
 
-// SetupWithManager TODO
+// SetupWithManager creates an setup a ManagedOCSReconciler to work with the provided manager
 func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	ctrlOptions := controller.Options{
 		MaxConcurrentReconciles: 1,
 	}
 	managedOCSPredicates := builder.WithPredicates(
 		predicate.GenerationChangedPredicate{},
 	)
-	addonParamsSecretWatchHandler := handler.EnqueueRequestsFromMapFunc{
+	addonParamsSecretPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(meta metav1.Object, _ runtime.Object) bool {
+				return meta.GetName() == r.AddonParamSecretName
+			},
+		),
+	)
+	monResourcesPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(meta metav1.Object, _ runtime.Object) bool {
+				labels := meta.GetLabels()
+				return labels == nil || labels[monLabelKey] != monLabelValue
+			},
+		),
+	)
+	enqueueManangedOCSRequest := handler.EnqueueRequestsFromMapFunc{
 		ToRequests: handler.ToRequestsFunc(
 			func(obj handler.MapObject) []reconcile.Request {
-				if obj.Meta.GetName() == r.AddonParamSecretName {
-					return []reconcile.Request{{
-						NamespacedName: types.NamespacedName{
-							Name:      "managedocs",
-							Namespace: obj.Meta.GetNamespace(),
-						},
-					}}
-				}
-				return nil
+				return []reconcile.Request{{
+					NamespacedName: types.NamespacedName{
+						Name:      "managedocs",
+						Namespace: obj.Meta.GetNamespace(),
+					},
+				}}
 			},
 		),
 	}
 
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(ctrlOptions).
 		For(&v1.ManagedOCS{}, managedOCSPredicates).
+
+		// Watch owned resources
 		Owns(&ocsv1.StorageCluster{}).
-		Watches(&source.Kind{Type: &corev1.Secret{}}, &addonParamsSecretWatchHandler).
+		Owns(&promv1.Prometheus{}).
+		Owns(&promv1.Alertmanager{}).
+		// Owns(&promv1a1.AlertmanagerConfig{}).
+
+		// Watch non-owned resources
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			&enqueueManangedOCSRequest,
+			addonParamsSecretPredicates,
+		).
+		Watches(
+			&source.Kind{Type: &promv1.PodMonitor{}},
+			&enqueueManangedOCSRequest,
+			monResourcesPredicates,
+		).
+		Watches(
+			&source.Kind{Type: &promv1.ServiceMonitor{}},
+			&enqueueManangedOCSRequest,
+			monResourcesPredicates,
+		).
+		Watches(
+			&source.Kind{Type: &promv1.PrometheusRule{}},
+			&enqueueManangedOCSRequest,
+			monResourcesPredicates,
+		).
+
+		// Create the controller
 		Complete(r)
 }
 
 // Reconcile TODO
 func (r *ManagedOCSReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("req.Namespace", req.Namespace, "req.Name", req.Name)
-	log.Info("Reconciling ManagedOCS")
+	log.Info("Starting reconcile for ManagedOCS")
 
-	r.ctx = context.Background()
-	r.namespace = req.Namespace
+	// Setup the reconciler from the request
+	r.reconileSetup(req)
```

Comment:
```misspelling
	r.reconcileSetup(req)
```

Reply:
I see the misspell, but in any case, I decided to rename it to `initReconciler`, which make more sense and also address the prev comment regarding the job of this function